tasks/cephfs/filesystem.py

```diff
@@ -738,7 +741,7 @@ def rados(self, args, pool=None, stdin_data=None):
 
         # NB we could alternatively use librados pybindings for this, but it's a one-liner
         # using the `rados` CLI
-        args = [os.path.join(self._prefix, "rados"), "-p", pool] + args
+        args = [os.path.join(self._prefix, "rados"), "-p", pool, "--namespace", namespace] + args
```

Comment:
This code should omit the namespace argument when it is None:
```
        args = ([os.path.join(self._prefix, "rados"), "-p", pool] +
                (["--namespace", namespace] if namespace else []) +
                args)
```

(and remove the part above where it changes None into "")